manifests/params.pp

```diff
@@ -42,44 +42,46 @@
       $package_version    = "${version_parts[0]}${version_parts[1]}"
 
       if $version == $postgresql::globals::default_version and $facts['os']['name'] != 'Amazon' or $postgresql::globals::manage_dnf_module {
-        $client_package_name    = pick($client_package_name, 'postgresql')
-        $server_package_name    = pick($server_package_name, 'postgresql-server')
-        $contrib_package_name   = pick($contrib_package_name,'postgresql-contrib')
-        $devel_package_name     = pick($devel_package_name, 'postgresql-devel')
-        $java_package_name      = pick($java_package_name, 'postgresql-jdbc')
-        $docs_package_name      = pick($docs_package_name, 'postgresql-docs')
-        $plperl_package_name    = pick($plperl_package_name, 'postgresql-plperl')
-        $plpython_package_name  = pick($plpython_package_name, 'postgresql-plpython')
-        $service_name           = pick($service_name, 'postgresql')
-        $bindir                 = pick($bindir, '/usr/bin')
-        $datadir                = $facts['os']['name'] ? {
+        $client_package_name     = pick($client_package_name, 'postgresql')
+        $server_package_name     = pick($server_package_name, 'postgresql-server')
+        $contrib_package_name    = pick($contrib_package_name,'postgresql-contrib')
+        $contrib_virtual_package = pick($contrib_virtual_package, undef)
```

Comment:
I recall that `pick()` has some problems with `pick($x, undef)`. I'm not sure this is valid.

Also, here it's guaranteed to have a value since `$contrib_package_name` is guaranteed to be a `String` at this point since `pick($x, 'MyString')` guarantees that.
```suggestion
```